File: backend/src/application/miners/mapper/miner_mapper.py
from backend.src.application.miners.dtos.miner_response_dto import MinerResponseDTO
from backend.src.core.miners.domain.entities.miner import Miner


class MinerMapper:

    @staticmethod
    def entity_to_dto(miner: Miner) -> MinerResponseDTO:
        return MinerResponseDTO(
            uuid=miner.uuid,
            name=miner.name,
            document_type=miner.document_type,
            document_number=miner.document_number,
            municipality=miner.municipality
        )

File: frontend/src/mappers/minerMapper.ts
import {Miner} from "../models/Miner.ts"
import {MinerDTO} from "../dtos/MinerResponseDTO.ts";

export class MinerMapper {

    static fromDTO(dto: MinerDTO) : Miner
    {
        return {
            uuid: dto.uuid,
            name: dto.name,
            documentType: dto.document_type,
            documentNumber: dto.document_number,
            municipality: dto.municipality
        }
    }

    static toDTO(miner: Miner) : MinerDTO
    {
        return {
            uuid: miner.uuid,
            name: miner.name,
            document_type: miner.documentType,
            document_number: miner.documentNumber,
            municipality: miner.municipality
        }
    }
}